enhancements/security/openstack-self-signed-certificates.md

```diff
@@ -0,0 +1,137 @@
+
+The `machine-controllers` and more specifically, `cluster-api-provider-openstack` (CAPO), will need to authenticate with openstack any time it needs to manage the current machine pool. In order to keep `openshift-config/cloud-provider-config` as the central point that services draw the CA certificate from, CAPO is given RBAC access to `GET` configmaps named `cloud-provider-config` from the `openshift-config` namespace. This allows the kubernetes client in CAPO to fetch the latest CA certificate, and craft a custom http transport that trusts it when authenticating with OpenStack.
+
+Lastly, Kuryr needs to communicate with OpenStack networking services as part of its core functionality. Following the common design pattern, kuryr GETs the latest version of `openshift-config/cloud-provider-config` and builds a custom http transport with the CA cert when authenticating with OpenStack.
+
+In total, these changes enable the distribution, and trust of OpenStack CA certificates in an OpenShift cluster both during and after install.
+
+## Motivation
+
+Numerous customer complaints have resulted in this issue being highly escalated. This prevents a number of customers from installing OpenShift 4 on their OpenStack clusters.
+
+### Goals
+
+1. To enable the OpenShift 4.4 installer to succesfully deploy a cluster on an OpenStack cloud that uses self signed certificates for auth.
+2. Openshift 4.4 can succesfully and stabily run on OpenStack clusters that use self signed certificates for auth.
+3. Backport to 4.3
+
+## Proposal
+
+### Risks and Mitigations
+
+1. It is possible to give services authorization/credentials that they dont need and shouldnt have, and this may pose a security risk.
+
+> To prevent this, we make sure that only services that need the CA cert get access to it, and the cert we pass them only has the CA cert in it.
+
+### Test Plan
+
+This feature should not affect any of the other platforms, so to ensure this we can take advantage of OpenShift's CI for those platforms. They should pass during a pull request, and their test pass rate should not change.
+
+For OpenStack, we have access to a cluster configured specifically to test this. We will be testing our code changes in that cluster before merging to ensure that the installer runs. Then, we can use the OpenShift CI and QE in order to garuntee that the code changes will not affect customers in any harmful ways.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- QE testing
+- Downstream Docs
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+### Upgrade / Downgrade Strategy
+
+This should not affect the Upgrade/Downgrade functionality.
```

Comment:
Well, it will break downgrade if you go from a 4.4 cluster that depends on this functionality to a 4.3 cluster that does not support it.  That's generally true of features that add core functionality, but maybe still worth calling out here.